File: easy/real_fake.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    next if ( $line =~ m/^\s$/ );
    chomp($line);
    $line =~ s/ //g;
    my @s = split( //, $line );
    for ( my $i = scalar @s - 2 ; $i >= 0 ; $i -= 2 ) {
        @s[$i] <<= 1;
    }
    my $r;
    map { $r += $_ } @s;
    print( ( $r % 10 == 0 ) ? "Real\n" : "Fake\n" );
}
close(INFILE);

File: easy/real_fake.rb
File.open(ARGV[0]).each_line do |line|
  s = line.delete(" \n").split('').map(&:to_i)
  (s.length - 2).step(0, -2) do |i|
    s[i] *= 2
  end
  puts s.inject(:+) % 10 == 0 ? 'Real' : 'Fake'
end
